### PR TITLE
test(type-utils): skip mysteriously failing type unit tests

### LINKED ISSUE #9426: Repo: v8 branch test failures on getTypeFlags and getTypeName: Cannot read properties of undefined (reading 'id')
### Suggestion

From e.g. https://cloud.nx.app/runs/95hV1MAyEv:

```plaintext
...
 FAIL  tests/typeFlagUtils.test.ts
  ● typeFlagUtils › getTypeFlags › when code is "type Test = any;", type flags is 1

    TypeError: Cannot read properties of undefined (reading 'id')

      19 |     const declaration = ast.body[0] as TSESTree.TSTypeAliasDeclaration;
      20 |
    > 21 |     return services.getTypeAtLocation(declaration.id);
         |                                                   ^
      22 |   }
      23 |
      24 |   describe('getTypeFlags', () => {

      at id (tests/typeFlagUtils.test.ts:21:51)
      at getType (tests/typeFlagUtils.test.ts:29:20)

 FAIL  tests/getTypeName.test.ts
  ● getTypeName › returns primitive type › when code is type Test = string;, returns string

    TypeError: Cannot read properties of undefined (reading 'id')

      20 |     const declaration = ast.body[0] as TSESTree.TSTypeAliasDeclaration;
      21 |
    > 22 |     return { checker, type: services.getTypeAtLocation(declaration.id) };
         |                                                                    ^
      23 |   }
      24 |
      25 |   function runTest(code: string, expected: string): void {

      at id (tests/getTypeName.test.ts:22:68)
      at getTypes (tests/getTypeName.test.ts:26:31)

Test Suites: 2 failed, 4 passed, 6 total
```

I'm confused by this. I merged the latest `main` from this repo into `v8`, but it's not happening for me locally. What are these test failures...?

🤷

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #9426
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Comments out the offending tests.

🤷

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -85,7 +85,7 @@
     "cross-fetch": "^4.0.0",
     "cspell": "^8.6.1",
     "downlevel-dts": ">=0.11.0",
-    "eslint": "8.57.0",
+    "eslint": "9",
```
Comment (JoshuaKGoldberg) [inline package.json:88]: I tried this too. No luck.

File: packages/type-utils/tests/getTypeName.test.ts [hunk 2]
```diff
@@ -30,7 +30,8 @@ describe('getTypeName', () => {
 
   describe('returns primitive type', () => {
     it.each([
-      ['type Test = string;', 'string'],
+      // TODO(#9426): re-enable this
+      // ['type Test = string;', 'string'],
       ['type Test = "text";', 'string'],
```
Comment (JoshuaKGoldberg) [inline packages/type-utils/tests/getTypeName.test.ts:35]: https://cloud.nx.app/runs/lXUjfK1oU8?utm_source=pull-request&utm_medium=comment:

```plaintext
 FAIL  tests/typeFlagUtils.test.ts
  ● typeFlagUtils › getTypeFlags › when code is "type Test = unknown;", type flags is 2

    TypeError: Cannot read properties of undefined (reading 'id')

      19 |     const declaration = ast.body[0] as TSESTree.TSTypeAliasDeclaration;
      20 |
    > 21 |     return services.getTypeAtLocation(declaration.id);
         |                                                   ^
      22 |   }
      23 |
      24 |   describe('getTypeFlags', () => {

      at id (tests/typeFlagUtils.test.ts:21:51)
      at getType (tests/typeFlagUtils.test.ts:29:20)

 FAIL  tests/getTypeName.test.ts
  ● getTypeName › returns primitive type › when code is type Test = "text";, returns string

    TypeError: Cannot read properties of undefined (reading 'id')

      20 |     const declaration = ast.body[0] as TSESTree.TSTypeAliasDeclaration;
      21 |
    > 22 |     return { checker, type: services.getTypeAtLocation(declaration.id) };
         |                                                                    ^
      23 |   }
      24 |
      25 |   function runTest(code: string, expected: string): void {

      at id (tests/getTypeName.test.ts:22:68)
      at getTypes (tests/getTypeName.test.ts:26:31)

Test Suites: 2 failed, 4 passed, 6 total
```

...so these are just altogether failing.

File: packages/type-utils/tests/getTypeName.test.ts [hunk 3]
```diff
@@ -30,7 +32,7 @@ describe('getTypeName', () => {
 
   describe('returns primitive type', () => {
     it.each([
-      ['type Test = string;', 'string'],
+      // ['type Test = string;', 'string'],
```
Comment (kirkwaiblinger) [inline packages/type-utils/tests/getTypeName.test.ts:35]: should this have the reenable todo as well?

**Review by kirkwaiblinger — APPROVED** [review]
ostritch algorithm

File: packages/type-utils/tests/getTypeName.test.ts (hunk 3 above)
Comment (JoshuaKGoldberg) [reply packages/type-utils/tests/getTypeName.test.ts:35]: Oh, I meant to un-comment this...

```suggestion
      ['type Test = string;', 'string'],
```

**Review by JoshuaKGoldberg — COMMENTED** [review]
Reporter sardonically saying "I hate myself" to an interview microphone, then walking away

**Comment by JoshuaKGoldberg** [comment]
Merging now to unblock the `v8` branch. Filed #9430 to follow up.

**Pull request merged**